xarray/plot/plot.py

```diff
@@ -384,8 +384,16 @@ def newplotfunc(darray, x=None, y=None, ax=None, row=None, col=None,
 
         _ensure_plottable(xval, yval)
 
-        if 'contour' in plotfunc.__name__ and levels is None:
-            levels = 7  # this is the matplotlib default
+        if 'contour' in plotfunc.__name__:
+            if levels is None:
+                levels = 7  # this is the matplotlib default
+            # A colorbar with one level is not possible with mpl
+            try:
```

Comment:
We could also change the default to `add_colorbar=None`, and only apply this logic if `add_colorbar` wasn't set directly.